### PR TITLE
Cleanup hab-butterfly crate

### PR DESCRIPTION
**Pull request by reset**
* Remove unused dependencies
* Fix compiler warnings
* Fix ergonomics of various public functions

tenor-209961408

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by mwrock** [comment]
oh looks like travis found a couple issues

**Comment by reset** [comment]
@mwrock this is dependent upon the other open PR I have. Will rebase and push when it's through the pipeline :)

File: components/hab-butterfly/Cargo.toml [hunk 1]
```diff
@@ -10,47 +10,13 @@ name = "hab-butterfly"
 doc = false
 
 [dependencies]
-clippy = {version = "*", optional = true}
-ansi_term = "*"
+clap = { version = "*", features = ["suggestions", "color", "unstable"] }
```
Comment (fnichol) [inline components/hab-butterfly/Cargo.toml]: There's the clap compile churner! You've saved us all compile time

**Pull request merged**